### ISSUE TITLE
Using class-based views with Flask 2.2 causes TypeError: view() takes 0 positional arguments but 1 was given

### ISSUE THREAD

**Issue by NimaQu** [apiflask/apiflask#341]
Error when using class-based views for compliance documentation

```
from flask.views import MethodView
from apiflask import APIFlask

app = APIFlask(__name__)


@app.route('/pets/<int:pet_id>', endpoint='pet')
class Pet(MethodView):

    def get(self, pet_id):
        return {'message': 'OK'}

    def delete(self, pet_id):
        return '', 204

app.run(debug=True)
```

Environment:

- Python version: 3.10
- Flask version: 2.2.2
- APIFlask version: 1.1.2

**Comment by hjlarry** [comment]
I compare the source code of `dispatch_request()` in flask and apiflask.
It seems this example class based routes only accepts keyword arguments.
The flask always return `view_function(**req.view_args)`.  
But for apiflask, I don't know why it return `view_function(*req.view_args.values())` for most situation

**Comment by greyli** [comment]
The change in https://github.com/pallets/flask/pull/4624 (Flask 2.2) caused this issue. I will try to fix this in Flask.

> @hjlarry I don't know why it return view_function(*req.view_args.values()) for most situation

It's used to make sure the arguments passed to the view function are in natural order ([docs](https://apiflask.com/comparison/#a-natural-order-of-view-arguments)).

**Comment by greyli** [comment]
I just released 1.1.3 to pin Flask < 2.2 as a temporal workaround for this issue. For the real fix to this issue, these are the possible solutions:

0. ~~Revert this change upstream (https://github.com/pallets/flask/pull/4776).~~
1. Patch the view function in our [add_url_rule method](https://github.com/apiflask/apiflask/blob/main/src/apiflask/route.py#L62). Not sure how to do this. Maybe [the Python signature](https://peps.python.org/pep-0362/) can do the trick.
2. Copy the whole `flask.view` module and modify it. This will be a breaking change since the user has to change the import statement from `from flask.views import MethodView` to `from apiflask.views import MethodView`.  And we have to mirror future changes for this module from Flask.
3. Turn all the arguments passed to the view function into keyword arguments ([proposed by davidism](https://github.com/pallets/flask/pull/4776#issuecomment-1222529526)). This seems a clean solution since we can remove the patched [`APIFlask.dispatch_request`](https://github.com/apiflask/apiflask/blob/main/src/apiflask/app.py#L392). This is also a breaking change. We have to set a default keyword for input data from different locations (`data` for JSON or `json` for JSON?). Users have to pass an argument to the `app.input` decorator to use a different argument name.

I prefer solution 1 now, while solution 3 seems a better option for the long term.

**Comment by hjlarry** [comment]
It seems solution 1 can compat with current user project, 2 and 3 are both broken change.
I think we can try whether solution  1 worked first,   2 and 3 we can concern for the next big version.
But I don't know how to patch a function.
